openmetadata-ui/src/main/resources/ui/src/components/Auth/AuthProviders/AuthProvider.tsx

```diff
@@ -356,55 +357,67 @@ export const AuthProvider = ({
   };
 
-  const handleSuccessfulLogin = (user: OidcUser) => {
+  const handleSuccessfulLogin = async (user: OidcUser) => {
     setLoading(true);
     setIsUserAuthenticated(true);
     const fields =
       authConfig?.provider === AuthProviderEnum.Basic
         ? userAPIQueryFields + ',' + isEmailVerifyField
         : userAPIQueryFields;
-    getLoggedInUser({ fields })
-      .then((res) => {
-        if (res) {
-          const updatedUserData = getUserDataFromOidc(res, user);
-          if (!matchUserDetails(res, updatedUserData, ['email'])) {
-            getUpdatedUser(updatedUserData, res);
-          } else {
-            setCurrentUser(res);
-          }
-
-          handledVerifiedUser();
-          // Start expiry timer on successful login
-          startTokenExpiryTimer();
-        }
-      })
-      .catch((err) => {
-        if (err && err.response && err.response.status === 404) {
-          setNewUserProfile(user.profile);
-          setCurrentUser({} as User);
-          setIsSigningIn(true);
-          history.push(ROUTES.SIGNUP);
+    try {
+      const res = await getLoggedInUser({ fields });
+      if (res) {
+        const updatedUserData = getUserDataFromOidc(res, user);
+        if (!matchUserDetails(res, updatedUserData, ['email'])) {
+          getUpdatedUser(updatedUserData, res);
         } else {
-          // eslint-disable-next-line no-console
-          console.error(err);
-          history.push(ROUTES.SIGNIN);
+          setCurrentUser(res);
         }
-      })
-      .finally(() => {
-        setLoading(false);
-      });
+
+        handledVerifiedUser();
+        // Start expiry timer on successful login
+        startTokenExpiryTimer();
+      }
+    } catch (error) {
+      const err = error as AxiosError;
+      if (err && err.response && err.response.status === 404) {
+        setNewUserProfile(user.profile);
+        setCurrentUser({} as User);
+        setIsSigningIn(true);
+        history.push(ROUTES.SIGNUP);
+      } else {
+        // eslint-disable-next-line no-console
+        console.error(err);
+        history.push(ROUTES.SIGNIN);
+      }
+    } finally {
+      setLoading(false);
+    }
   };
 
   const handleSuccessfulLogout = () => {
     resetUserDetails();
   };
 
+  /**
+   * Stores redirect URL for successful login
+   */
+  const handleStoreProtectedRedirectPath = useCallback(() => {
+    if (isProtectedRoute(location.pathname)) {
+      storeRedirectPath(location.pathname);
+    }
+  }, [location.pathname, storeRedirectPath]);
+
   const updateAuthInstance = (configJson: AuthenticationConfiguration) => {
     const { provider, ...otherConfigs } = configJson;
     switch (provider) {
       case AuthProviderEnum.Azure:
         {
-          setMsalInstance(otherConfigs as unknown as Configuration);
+          setMsalInstance(
+            new PublicClientApplication(
+              otherConfigs as unknown as Configuration
+            )
+          );
```

Comment:
Have we tested these changes?

Reply:
Yes we have tested the Azure SSO and this has been covered there.